src/main/java/com/odysseusinc/arachne/portal/api/v1/controller/BaseAdminController.java

```diff
@@ -146,6 +157,50 @@ public void fullDeleteStudy(Long[] ids) {
         return users.map(user -> conversionService.convert(user, CommonUserDTO.class));
     }
 
+    @ApiOperation("Register new users")
+    @RequestMapping(value = "/api/v1/admin/users/registration", method = RequestMethod.POST)
+    public void register(@RequestBody BulkUsersRegistrationDTO bulkUsersDto) throws PasswordValidationException, ValidationException {
+
+        if (bulkUsersDto.getTenants() == null || bulkUsersDto.getTenants().isEmpty()) {
+            throw new ValidationException("tenants: must be not empty");
+        }
+
+        boolean emailConfirmationRequired = bulkUsersDto.getEmailConfirmationRequired();
+
+        Set<Tenant> tenants = bulkUsersDto.getTenants().stream()
+                .map(tenant -> conversionService.convert(tenant, Tenant.class))
+                .collect(Collectors.toSet());
+        List<IUser> users = bulkUsersDto.getUsers().stream()
```

Comment:
It's better to use generic for IUser like in BaseUserController done and then typecasts will gone.